EstateReporting.IntegrationTests/Common/DockerHelper.cs

```diff
@@ -0,0 +1,503 @@
+        /// <summary>
+        /// The transaction processor client
+        /// </summary>
+        public ITransactionProcessorClient TransactionProcessorClient;
+
+        /// <summary>
+        /// The estate reporting client
+        /// </summary>
+        public IEstateReportingClient EstateReportingClient;
+
+        /// <summary>
+        /// The containers
+        /// </summary>
+        protected List<IContainerService> Containers;
+
+        /// <summary>
+        /// The estate management API port
+        /// </summary>
+        protected Int32 EstateManagementApiPort;
+
+        /// <summary>
+        /// The event store HTTP port
+        /// </summary>
+        protected Int32 EventStoreHttpPort;
+
+        /// <summary>
+        /// The security service port
+        /// </summary>
+        protected Int32 SecurityServicePort;
+
+        /// <summary>
+        /// The test networks
+        /// </summary>
+        protected List<INetworkService> TestNetworks;
+
+        protected String SecurityServiceContainerName;
+
+        protected String EstateManagementContainerName;
+
+        protected String EventStoreContainerName;
+
+        protected String EstateReportingContainerName;
+
+        protected String SubscriptionServiceContainerName;
+
+        protected String TransactionProcessorContainerName;
+
+        protected String TestHostContainerName;
+
+        /// <summary>
+        /// The transaction processor port
+        /// </summary>
+        protected Int32 TransactionProcessorPort;
+
+
+        protected Int32 EstateReportingPort;
+
+        /// <summary>
+        /// The logger
+        /// </summary>
+        private readonly NlogLogger Logger;
+
+        private readonly TestingContext TestingContext;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DockerHelper" /> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="testingContext">The testing context.</param>
+        public DockerHelper(NlogLogger logger, TestingContext testingContext)
+        {
+            this.Logger = logger;
+            this.TestingContext = testingContext;
+            this.Containers = new List<IContainerService>();
+            this.TestNetworks = new List<INetworkService>();
+        }
+
+        #endregion
+
+        public const Int32 TestHostPort = 9000;
+
+        public static IContainerService SetupTestHostContainer(String containerName, ILogger logger, String imageName,
+                                                               List<INetworkService> networkServices,
+                                                               String hostFolder,
+                                                               (String URL, String UserName, String Password)? dockerCredentials,
+                                                               Boolean forceLatestImage = false)
```

Comment:
Codacy found an issue: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/stuart_ferguson1/EstateReporting/pullRequest?prid=6339123)